miden-lib/asm/note_scripts/basic.masm

```diff
@@ -0,0 +1,75 @@
+use.miden::sat::account
+use.miden::sat::note
+use.miden::sat::tx
+use.miden::wallets::basic->wallet
+
+#! Pay to ID - ensures that note can only be consumed by an account with the provided ID.
+#! 
+#! Inputs: [target_account_id]
+#! Outputs: []
+#!
+#! - ASSET is the asset to be received, can be fungible or non-fungible
+#!
+#! FAILS if: 
+#! - Account ID of executing account is not equal to specified Account ID
+#! - The same non-fungible asset already exists in the account.
+#! - Adding a fungible asset would result in amount overflow, i.e.,
+#!   the total amount would be greater than 2^63.
+export.p2id                                             # [note_inputs = target_account_id, ...]                                            
+    exec.account::get_id                                # [account_id, target_account_id, ...]
+    
+    assert_eq                                           # [] if account_id == target_account_id, fails if not
+
+    push.0                                              # [0, ...] memory pointer to store assets
+    exec.note::get_assets                               # [num_of_assets, 0, ...] 
+
+    dup neq.0                                           # [1 || 0, num_of_assets, 0, ...], sanity check that there is at least one asset
+    while.true                                          
+        sub.1 padw swap.5 dup swap.6                    # [num_of_assets - (i), 0, 0, 0, 0, num_of_assets - (i), ...], i ~ round in the loop
+        mem_loadw                                       # [ASSET, num_of_assets - (i), ...], i ~ round in the loop
+        call.wallet::receive_asset dropw                # [num_of_assets - (i), ...], i ~ round in the loop
+        dup neq.0                                       # [1 || 0, num_of_assets - (i), ...], if end of memory reached, break
+    end
+
+    drop drop
+end
```

Comment:
Could we use the same commenting style as we use in other places - not only for doc comments but for inline comments as well (e.g., [here](https://github.com/0xPolygonMiden/miden-vm/blob/main/stdlib/asm/collections/smt.masm#L55))?

Also, we can probably optimize the loop a bit more:
- Instead of using `sub.1` (which takes 4 cycles) we can negate the counter before we enter the loop and then use `add.1` (which takes 1 cycle).
- We should be able to avoid using `padw` and `dropw` inside the loop as ` mem_loadw` overwrites w/e is on the top of the stack.

Reply:
> Instead of using sub.1 (which takes 4 cycles) we can negate the counter before we enter the loop and then use add.1 (which takes 1 cycle).

Changed the loop to use add. Not sure about the cycle count now, it would be nice to be able to measure it, but I can't run it in isolation. 

> We should be able to avoid using padw and dropw inside the loop as  mem_loadw overwrites w/e is on the top of the stack.

I played around with it, but I think we need to keep `padw` and `dropw` exactly because `mem_load` overwrites the top four items. Without `padw` we would lose our counter and the final memory position.